### PR TITLE
API: Add compression argument to Series.to_csv

### LINKED ISSUE #18958: pd.Series to_csv method does not recognize "compression"
There looks like something is wrong with the `compression` argument in `to_csv` for`pd.Series` but `pd.DataFrame`.  This error is also in 0.20.3.
```python
In [1]: Se_tmp = pd.Series(list("ACGT"))
   ...: Se_tmp.to_csv("~/test.tsv.gz", sep="\t", compression="gzip")
   ...:
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-de7ce483228f> in <module>()
      1 Se_tmp = pd.Series(list("ACGT"))
----> 2 Se_tmp.to_csv("~/test.tsv.gz", sep="\t", compression="gzip")

TypeError: to_csv() got an unexpected keyword argument 'compression'

In [2]: Se_tmp.to_frame("testing_column").to_csv("~/test.tsv.gz", sep="\t", compression="gzip")

In [3]: pd.__version__
Out[3]: '0.21.1'
```

### PR DESCRIPTION
**Pull request by reidy-p**
- [ ] closes #18958
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
see comments

**Comment by reidy-p** [comment]
@jreback did you post comments? I can't see any.

**Comment by jreback** [comment]
sorry, see #19226 want to parametrize these tests

File: pandas/tests/series/test_io.py [hunk 1]
```diff
@@ -142,12 +139,31 @@ def test_to_csv_path_is_none(self):
         csv_str = s.to_csv(path=None)
         assert isinstance(csv_str, str)
 
-    @pytest.mark.parametrize('compression, open_func', [
-        ('gzip', gzip.open),
-        ('bz2', bz2.BZ2File),
-        (pytest.param('xz', lzma.open, marks=td.skip_if_no_lzma)),
+    def decompress_file(self, src_path, compression):
```
Comment (jreback) [inline pandas/tests/series/test_io.py]: can u move this function to pandas.util.testing as it’s more generally useful for testing

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by gfyoung** [comment]
@jreback @reidy-p : I like the `decompress` utility (it should be applied to our `DataFrame.to_csv` tests), but this patch for `Series.to_csv` is only a band-aid.  We should implement what you summarized in your comment <a href="https://github.com/pandas-dev/pandas/issues/18958#issuecomment-357449553">here</a> to close this issue.

**Comment by jreback** [comment]
@gfyoung I agree, but it puts this in a central location at least for now.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @reidy-p ok would love to work on centralizing the compression tests!

**Pull request merged**